plugins/external_setup_hook/app/models/external_setup_hook.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class ExternalSetupHook < ActiveRecord::Base
+  validates :name, presence: true
+
+  validate :validate_endpoint
+
+  has_many :stage_external_setup_hooks, dependent: :destroy
+  has_many :stages, through: :stage_external_setup_hooks, dependent: :destroy
+
+  def validate_endpoint
+    errors.add(:auth_type, "is unsupported") if auth_type.downcase! != 'token'
```

Comment:
I'd prefer not to downcase ... better have ppl get this right instead of guessing and then ending up with strange errors